gateway/ln-gateway/src/lib.rs

```diff
@@ -442,40 +443,59 @@ impl Gateway {
             lightning_alias: _,
         } = self.state.read().await.clone()
         {
-            while let Some(Ok(htlc_request)) = stream.next().await {
-                if handle.is_shutting_down() {
-                    break;
-                }
+            loop {
+                match stream.next().await {
+                    Some(Ok(htlc_request)) => {
+                        if handle.is_shutting_down() {
+                            break;
+                        }
 
-                let scid_to_feds = self.scid_to_federation.read().await;
-                let federation_id = scid_to_feds.get(&htlc_request.short_channel_id);
-                // Just forward the HTLC if we do not have a federation that
-                // corresponds to the short channel id
-                if let Some(federation_id) = federation_id {
-                    let clients = self.clients.read().await;
-                    let client = clients.get(federation_id);
-                    // Just forward the HTLC if we do not have a client that
-                    // corresponds to the federation id
-                    if let Some(client) = client {
-                        let htlc = htlc_request.clone().try_into();
-                        if let Ok(htlc) = htlc {
-                            if client.gateway_handle_intercepted_htlc(htlc).await.is_ok() {
-                                continue;
+                        let scid_to_feds = self.scid_to_federation.read().await;
+                        let federation_id = scid_to_feds.get(&htlc_request.short_channel_id);
+                        // Just forward the HTLC if we do not have a federation that
+                        // corresponds to the short channel id
+                        if let Some(federation_id) = federation_id {
+                            let clients = self.clients.read().await;
+                            let client = clients.get(federation_id);
+                            // Just forward the HTLC if we do not have a client that
+                            // corresponds to the federation id
+                            if let Some(client) = client {
```

Comment:
Maybe we can split some of these `if let Some` patterns into helper functions to help with indentation here too

Reply:
Not feeling very creative today. Although the diff is shitty, the final code looks fine.

Reply:
Let's do this in a follow up